server/ControlPlane/ServiceMetadata/ServiceMetadata.cs

```diff
@@ -3,29 +3,44 @@
 
 using Microsoft.Extensions.Options;
 using Tyger.ControlPlane.Auth;
+using Tyger.ControlPlane.Versioning;
 
 namespace Tyger.ControlPlane.ServiceMetadata;
 
 public static class ServiceMetadata
 {
-    public static void MapServiceMetadata(this WebApplication app)
+    public static void MapServiceMetadata(this RouteGroupBuilder root)
     {
         Model.ServiceMetadata? serviceMetadata = null;
-        app.MapGet(
-            "/v1/metadata",
+        root.MapGet(
+            "/metadata",
             (IEnumerable<ICapabilitiesContributor> contributor, IOptions<AuthOptions> auth) =>
             {
                 if (serviceMetadata is null)
                 {
                     var capabilities = contributor.Aggregate(Capabilities.None, (acc, c) => acc | c.GetCapabilities());
                     var capabilityStrings = Enum.GetValues<Capabilities>().Where(c => c != Capabilities.None && capabilities.HasFlag(c)).Select(c => c.ToString()).ToList();
-                    serviceMetadata = auth.Value.Enabled
-                        ? new(auth.Value.Authority, auth.Value.Audience, auth.Value.CliAppUri, Capabilities: capabilityStrings)
-                        : new(Capabilities: capabilityStrings);
+                    var apiVersionsSupported = ApiVersions.SupportedVersions().Select(v => v.ToString()).ToList();
+
+                    serviceMetadata = new Model.ServiceMetadata
+                    {
+                        Capabilities = capabilityStrings,
+                        ApiVersions = apiVersionsSupported
+                    };
+
+                    if (auth.Value.Enabled)
+                    {
+                        serviceMetadata = serviceMetadata with
+                        {
+                            Authority = auth.Value.Authority,
+                            Audience = auth.Value.Audience,
+                            CliAppUri = auth.Value.CliAppUri
+                        };
+                    }
                 }
 
                 return serviceMetadata;
-            }).AllowAnonymous();
+            }).AllowAnonymous().IsApiVersionNeutral();
```

Comment:
Hitting the metadata endpoint with an unsuported api-version should give a bad request, but instead returns 401 Unauthorized, unless a valid token is given.

```
$ curl 'http://localhost:5000/metadata?api-version=1.1' -i
HTTP/1.1 401 Unauthorized
Content-Length: 0
Date: Thu, 10 Apr 2025 23:38:29 GMT
Server: Kestrel
WWW-Authenticate: Bearer authority=https://login.microsoftonline.com/76d3279b-830e-4bea-baf8-12863cdeba4c, audience=api://tyger-server
Request-Id: 0HNBOH0A0HM8S:00000001
```